Sources/Attribution/AttributionPoster.swift

```diff
@@ -88,46 +87,27 @@ class AttributionPoster {
         }
 
         if !newData.isEmpty {
-            if network == .appleSearchAds {
-                postSearchAds(newData: newData,
-                              network: network,
-                              appUserID: currentAppUserID,
-                              newDictToCache: newDictToCache)
-            } else {
-                postSubscriberAttributes(newData: newData,
-                                         network: network,
-                                         appUserID: currentAppUserID,
-                                         newDictToCache: newDictToCache)
-            }
+            postSubscriberAttributes(newData: newData,
+                                     network: network,
+                                     appUserID: currentAppUserID,
+                                     newDictToCache: newDictToCache)
         }
     }
 
-    func postAppleSearchAdsAttributionIfNeeded() {
-        guard attributionFetcher.isAuthorizedToPostSearchAds else {
+    // should match OS availability in https://developer.apple.com/documentation/ad_services
+    @available(iOS 14.3, macOS 11.1, macCatalyst 14.3, *)
+    func postAdServicesTokenIfNeeded() {
+        let latestTokenSent = latestNetworkIdAndAdvertisingIdentifierSent(network: .adServices)
+        guard latestTokenSent == nil else {
             return
         }
 
-        let latestIdsSent = latestNetworkIdAndAdvertisingIdentifierSent(network: .appleSearchAds)
-        guard latestIdsSent == nil else {
+        guard let attributionToken = attributionFetcher.adServicesToken else {
             return
         }
 
-        attributionFetcher.afficheClientAttributionDetails { attributionDetails, error in
-            guard let attributionDetails = attributionDetails,
-                  error == nil else {
-                return
-            }
-
-            let attributionDetailsValues = Array(attributionDetails.values)
-            let firstAttributionDict = attributionDetailsValues.first as? [String: NSObject]
-
-            guard let hasIad = firstAttributionDict?["iad-attribution"] as? NSNumber,
-                  hasIad.boolValue == true else {
-                return
-            }
-
-            self.post(attributionData: attributionDetails, fromNetwork: .appleSearchAds, networkUserId: nil)
-        }
+        Logger.debug("Logging attribution token for now to avoid lint warning: \(attributionToken)")
+        // post
```

Comment:
Maybe leave this as a `TODO`? it's okay if lint fails since we're not merging this in `main` yet. That way we ensure it doesn't slip through the cracks.